safer_remote_clusters.md

```diff
@@ -0,0 +1,43 @@
+# Safer Remote Clusters (proposal)
+
+### Problem
+
+Users often change their kube context, forget about it, `tilt up`, and end up
+deploying to prod.
+
+### Solutions
+
+The following are not mutually exclusive!
+
+#### Point users at a pure Starlark Solution
+
+a la [what dmiller came up with](https://gist.github.com/jazzdan/d30ef73482920beb3ac9fa2fcb2c3146)
+
+Pros:
+* cheap
+* flexible
+
+Cons:
+* not very discoverable
+* unsafe by default
+
+#### Require intervention when deploying over a non-Tilt-deployed entity
+
+Pros:
+* would be nice to have regardless of anything else we do
+
+Cons:
+* not clear we have a good existing UI pattern for such an intervention
+* doesn't protect users from deploying non-prod resources to prod (e.g., I've `tilt up`'d both integration tests and servantes in prod this week), though that's obviously a much less dire problem than killing production
+
+#### Require remote k8s clusters to be explicitly whitelisted
+
+Add an `allowed_remote_k8s_clusters` Tiltfile directive. Generate a Tiltfile error unless either:
+1. the k8s context api's host is localhost
+2. the k8s context's name is passed to `allowed_remote_k8s_clusters`
```

Comment:
1) s/clusters/contexts/ ?
2) s/_remote// ? I don't want the user to be mad that we say it's remote, even though it's just that you don't realize my context is local, so I have to pass it.
3) Allow user to turn off the default, so they can make sure not to deploy to microk8s for some reason. Maybe via a vararg like disable_default_local_clusters

Reply:
these all sound good